### ISSUE TITLE
Any advisement to get your result?

### ISSUE THREAD

**Issue by xmyqsh** [StanfordASL/Trajectron-plus-plus#13]
Hello,
       I got 8th(last) place in the [Prediction Challenge](https://evalai.cloudcv.org/web/challenges/challenge-page/591/leaderboard/1659#leaderboardrank-8) by using this repo method, while your official result is 3th place.

```
rand (TPP-LSTM-GRAPH-CVAE-GMM) | 10.06 | 9.72 | 0.86 | 0.82 | 15.66 | 0.37

StanfordASL (Trajectron++) | 1.88 | 1.51 | 0.70 | 0.57 | 9.52 | 0.25
```

Any advisement to get your result?

By the way, I notice lots of cases in my result that the best of the generated trajectory's probability is lower than 0.5, even approaching zero. While I also notice lots of good trajectories whose probability is 1.

By the way, what I noticed is that almost all the low probability best trajectories, their first predicted point's probability is approaching zero. And there are lots of such trajectories in my prediction.

And, I observed a similar conclusion by using the official model which is trained with 8 history point and predict 6 future point.

By the way, I use `def p_y_xz(` to calculate the conditional point probability.

Some key hyperparams:
```
min_history_points: 4
prediction_horizon: 12
model: int_ee_me
```

**Comment by BorisIvanovic** [comment]
The main changes we made from this codebase were:

1. How we extracted data in `experiments/nuscenes/process_data.py`

2. How we predicted trajectories. We used the following call to the model's `predict` function:
```
probs, predictions = trajectron.predict(test_scene,
                                                  np.array([timestep]),
                                                  ph,
                                                  num_samples=1,
                                                  all_z_sep=True,
                                                  gmm_mode=True,
                                                  full_dist=False)
```
where the mode probabilities can be obtained by switching the last line in `mgcvae.py` to
```
            return self.latent.p_dist.probs, our_sampled_future
```
For a bit more information, take a look [this question](https://github.com/StanfordASL/Trajectron-plus-plus/issues/11) I responded to, there I write about how to set the model up to make predictions for the nuScenes prediction challenge.

3. We also used the `int_ee_me` model, lowered its initial learning rate to 0.002, and retrained it on the train set provided by the nuScenes prediction challenge (using 4 history / 12 prediction timesteps).

**Comment by xmyqsh** [comment]
All right, `lower its initial learning rate to 0.002`, because half of the history, from 8 to 4, is used.

Thanks for your reply, I will have a try.

**Comment by xmyqsh** [comment]
Honestly, I haven't examined its performance in the training set...
